python/tests/expressions/test_evaluator.py

```diff
@@ -0,0 +1,203 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+from typing import Any, List, TypeVar
+
+from pyiceberg.expressions import AlwaysTrue, AlwaysFalse, LessThan, Reference, LessThanOrEqual, GreaterThan, \
+    GreaterThanOrEqual, EqualTo, NotEqualTo, IsNaN, NotNaN, IsNull, NotNull, Not, And, Or, In, NotIn
+from pyiceberg.expressions.literals import literal
+from pyiceberg.expressions.visitors import evaluator
+from pyiceberg.files import StructProtocol
+from pyiceberg.schema import Schema
+from pyiceberg.types import NestedField, LongType, StringType, DoubleType
+
+V = TypeVar('V')
+
+
+class Column:
+    """A helper class for creating expressions"""
+    name: str
+
+    def __init__(self, name):
+        self.name = name
+
+    def __lt__(self, other: V) -> LessThan:
+        return LessThan(term=Reference(self.name), literal=literal(other))
+
+    def __le__(self, other: Any) -> LessThanOrEqual:
+        return LessThanOrEqual(term=Reference(self.name), literal=literal(other))
+
+    def __gt__(self, other: Any) -> GreaterThan:
+        return GreaterThan(term=Reference(self.name), literal=literal(other))
+
+    def __ge__(self, other: Any) -> GreaterThanOrEqual:
+        return GreaterThanOrEqual(term=Reference(self.name), literal=literal(other))
+
+    def __eq__(self, other: Any) -> EqualTo:
+        return EqualTo(term=Reference(self.name), literal=literal(other))
+
+    def __ne__(self, other: Any) -> NotEqualTo:
+        return NotEqualTo(term=Reference(self.name), literal=literal(other))
+
+    def is_null(self) -> IsNaN:
+        return IsNull(term=Reference(self.name))
+
+    def not_null(self) -> IsNaN:
+        return NotNull(term=Reference(self.name))
+
+    def is_nan(self) -> IsNaN:
+        return IsNaN(term=Reference(self.name))
+
+    def not_nan(self) -> IsNaN:
+        return NotNaN(term=Reference(self.name))
+
+
+def col(name: str) -> Column:
+    return Column(name)
+
+
+class Record(StructProtocol):
```

Comment:
@Fokko, I also had to add this class since we don't really have a generic record class anywhere. We should follow up and move this into a `data` package or something.